src/js/components/Select/SelectContainer.js

```diff
@@ -136,7 +136,12 @@ const SelectContainer = forwardRef(
         const clearButton = clearRef.current;
         if (onSearch) {
           const searchInput = searchRef.current;
-          if (searchInput && searchInput.focus && !activeRef.current) {
+          if (
+            searchInput &&
+            searchInput.focus &&
+            !activeRef.current &&
+            document.activeElement !== clearButton
```

Comment:
This additional condition is intended to prevent an unwanted refocus of the search input when the Select dropdown is triggered by a click, then a user initiates keyboard navigation.

Without it, we'd be back at a similar situation I fixed in #6982.

Only `SelectOption` elements are considered for `activeRef`.